lib/jekyll-paginate-v2/generator/paginator.rb

```diff
@@ -20,42 +20,60 @@ def initialize(config_per_page, first_index_page_url, paginated_page_url, posts,
         @per_page = config_per_page.to_i
         @total_pages = num_pages
 
-        if @page > @total_pages
-          raise RuntimeError, "page number can't be greater than total pages: #{@page} > #{@total_pages}"
-        end
+        # RuntimeError is raised by default
+        raise "page number can't be greater than total pages: #{@page} > #{@total_pages}" if @page > @total_pages
+
+        init  = (@page - 1) * @per_page
+        count = init + @per_page - 1
+
+        # get the smaller of two values
+        offset = [count, posts.size].min
 
-        init = (@page - 1) * @per_page
-        offset = (init + @per_page - 1) >= posts.size ? posts.size : (init + @per_page - 1)
+        # Ensure that default attributes are neither `nil` nor `false`
+        default_indexpage ||= ''
+        default_ext ||= ''
 
         # Ensure that the current page has correct extensions if needed
-        this_page_url = Utils.ensure_full_path(@page == 1 ? first_index_page_url : paginated_page_url,
-                                               !default_indexpage || default_indexpage.length == 0 ? 'index' : default_indexpage,
-                                               !default_ext || default_ext.length == 0 ? '.html' : default_ext)
+        page_url = @page == 1 ? first_index_page_url : paginated_page_url
+        basename = default_indexpage.empty? ? 'index' : default_indexpage
+        ext_name = default_ext.empty? ? '.html' : default_ext
+
+        this_page_url = Utils.ensure_full_path(page_url, basename, ext_name)
         
         # To support customizable pagination pages we attempt to explicitly append the page name to 
-        # the url incase the user is using extensionless permalinks. 
-        if default_indexpage && default_indexpage.length > 0
-          # Adjust first page url
+        # the url in case the user is using extensionless permalinks.
+        unless default_indexpage.empty?
           first_index_page_url = Utils.ensure_full_path(first_index_page_url, default_indexpage, default_ext)
-          # Adjust the paginated pages as well
-          paginated_page_url = Utils.ensure_full_path(paginated_page_url, default_indexpage, default_ext)
+          paginated_page_url   = Utils.ensure_full_path(paginated_page_url, default_indexpage, default_ext)
         end        
 
         @total_posts = posts.size
-        @posts = posts[init..offset]
-        @page_path = Utils.format_page_number(this_page_url, cur_page_nr, @total_pages)
+        @posts       = posts[init..offset]
+        @page_path   = _number_page(this_page_url, cur_page_nr)
```

Comment:
Please retain the name of the Utils function if you want to introduce a wrapper, eg `_format_page_number` or alternatively keep the part of the name that describes the best what its purpose is, in this case I'd be happy with `_format`

Reply:
renamed to `_format_page_number`